### ISSUE TITLE
Example of jitpack use

### ISSUE THREAD

**Issue by 0x6e6562** [tbroyer/gradle-apt-plugin#83]
What's the idiomatic way of using a jitpack version of the plugin?

For example

```
plugins {
   id "com.github.tbroyer.gradle-apt-plugin" version "master-v0.15-g0561c08-11"
}
```

Doesn't appear to resolve even when `settings.gradle` contains

```
pluginManagement {
  repositories {
    maven { url 'https://jitpack.io' }
    gradlePluginPortal()
  }
}
```

Am I using the wrong naming scheme for the plugin?

**Comment by 0x6e6562** [comment]
Playing around with this a bit more, I've discovered that 

```
dependencies {
  implementation 'com.github.tbroyer:gradle-apt-plugin:0561c08'
  ...
}
```

appears to trigger a jitpack build, but in this instance, the actual jitpack build is failing:

https://jitpack.io/com/github/tbroyer/gradle-apt-plugin/0561c08/build.log

**Comment by tbroyer** [comment]
The ID of the plugin is still `net.ltgt.apt`, but the way JitPack works this cannot work the way Gradle looks for the [plugin marker artifact](https://docs.gradle.org/current/userguide/plugins.html#sec:plugin_markers).

You need to use the "old way", using `buildscript` and `apply`.

**Comment by 0x6e6562** [comment]
Many thanks for the heads up - do you have an example for `buildscript` and `apply`?

**Comment by tbroyer** [comment]
https://plugins.gradle.org/plugin/net.ltgt.apt :wink:

**Comment by 0x6e6562** [comment]
Cool - is there a way to get a build for the `0561c08` commit? Or is it best to try to build and install it locally?

For reference, I'm trying to debug a spurious issue with the 0.15 release that causes the immutables.org annotations to get ignored, so I was trying to see whether I can reproduce the issue with the latest master version of the plugin.

**Comment by tbroyer** [comment]
Wrt Jitpack, I suppose it doesn't detect the `build.gradle.kts` and maybe creates an almost-empty `build.gradle` file; and that would have indeed started breaking between 0.14 and 0.15 when I switched to the Gradle Kotlin DSL.

Was there something in partticular you wanted to use from the latest snapshot that's not in 0.15?

**Comment by 0x6e6562** [comment]
The issue I'm seeing with 0.15 is only reproducible on one environment, I do not see this in CI, so I thought that using a snapshot might have a different behavior.

Unless there is debug flag specific to this plugin that can increase the verbosity just for annotation processing?

**Comment by tbroyer** [comment]
Which version of Gradle are you using? If you're using a recent version of Gradle (4.6 or 4.7), the plugin actually does almost nothing.
Do you see the correct `-processorpath` on each task when building with `--debug` logs, and no `-proc:none`?

**Comment by tbroyer** [comment]
If you want to debug a local version, I think the best is to use Gradle's `--include-build` [composite build feature](https://docs.gradle.org/current/userguide/composite_builds.html) (that would mean using the `buildscript` way too; see limitation at end of the composite builds doc page)

**Comment by 0x6e6562** [comment]
I'm using 4.7.

I'll look into the `-processorpath` now.

**Comment by tbroyer** [comment]
Fyi, I removed the jitpack stance from the README. If someone knows how to fix the builds there, please send a PR (apparently JitPack works OK for other people using the Gradle Kotlin DSL.

I also released a v0.16 yesterday, but be aware that it introduces unexpected breaking changes to Eclipse (and to a lesser extent to IntelliJ IDEA too). I'll try to work on a 0.17 soon, but in the mean time you might want to stick to 0.15.

**Comment by 0x6e6562** [comment]
Many thanks for the update
